src/utils/validateBooking.ts

```diff
@@ -0,0 +1,63 @@
+import moment from "moment";
+import { Booking } from "../generated/prisma-client";
+import { Context } from "prisma-client-lib/dist/types";
+
+type BookingData = {
+  start: Date;
+  end: Date;
+};
+
+const getTimeSlots = (start: Date, end: Date): Date[] => {
+  const timeSlots: Date[] = [];
+  const duration = moment(end).diff(moment(start), "hour");
+  for (let i = 0; i < duration; i++) {
+    const timeSlot = moment(start)
+      .add(i, "hour")
+      .toDate();
+    timeSlots.push(timeSlot);
+  }
+  return timeSlots;
+};
+
+export const validateBooking = async (
+  roomNumber: string,
+  booking: BookingData,
+  ctx
```

Comment:
unable to bind type definitions for `ctx: Context` without producing errors
`ctx.prisma.room()` asks for 3 arguments, and this function ends up pointing towards the custom resolver we have for `Booking { room (parent, arg, ctx) }` function
when `Context` definition is not used, this function points towards the db querying function generated by prisma (this is what we want)